test/e2e/test_broker_pubsub.go

```diff
@@ -0,0 +1,185 @@
+		"brokerName":       brokerName,
+		"dummyTriggerName": dummyTriggerName,
+		"respTriggerName":  respTriggerName,
+		"kserviceName":     kserviceName,
+		"senderName":       senderName,
+		"targetName":       targetName,
+		"clusterRoleName":  clusterRoleName,
+	}
+	for k, v := range packages {
+		config[k] = v
+	}
+
+	// Create resources
+	brokerInstaller := createresource(client, config, []string{"pubsub_broker", "istio"}, t)
+	defer deleteResource(brokerInstaller, t)
+
+	// Wait for broker, trigger, ksvc ready
+	brokerGVR := schema.GroupVersionResource{
+		Group:    "eventing.knative.dev",
+		Version:  "v1alpha1",
+		Resource: "brokers",
+	}
+	if err := client.WaitForResourceReady(client.Namespace, brokerName, brokerGVR); err != nil {
+		t.Error(err)
+	}
+
+	triggerGVR := schema.GroupVersionResource{
+		Group:    "eventing.knative.dev",
+		Version:  "v1alpha1",
+		Resource: "triggers",
+	}
+
+	if err := client.WaitForResourceReady(client.Namespace, dummyTriggerName, triggerGVR); err != nil {
+		t.Error(err)
+	}
+	if err := client.WaitForResourceReady(client.Namespace, respTriggerName, triggerGVR); err != nil {
+		t.Error(err)
+	}
+
+	ksvcGVR := schema.GroupVersionResource{
+		Group:    "serving.knative.dev",
+		Version:  "v1alpha1",
+		Resource: "services",
+	}
+	if err := client.WaitForResourceReady(client.Namespace, kserviceName, ksvcGVR); err != nil {
+		t.Error(err)
+	}
+
+	// Get broker URL
+	metaAddressable := resources.NewMetaResource(brokerName, client.Namespace, common.BrokerTypeMeta)
+	u, err := base.GetAddressableURI(client.Dynamic, metaAddressable)
+	if err != nil {
+		t.Error(err.Error())
+	}
+	config["brokerURL"] = u.String()
+
+	time.Sleep(5 * time.Second)
+
+	// Send a dummy cloudevent to broker
+	senderInstaller := createresource(client, config, []string{"pubsub_sender"}, t)
+	defer deleteResource(senderInstaller, t)
+
+	jobGVR := schema.GroupVersionResource{
+		Group:    "batch",
+		Version:  "v1",
+		Resource: "jobs",
+	}
+	// Check if dummy cloudevent is sent out
+	if done := jobDone(client, senderName, t, jobGVR); !done {
+		t.Error("dummy event didn't send to broker")
+		t.Failed()
+	}
+	// Check if resp cloudevent hits the target service
+	if done := jobDone(client, targetName, t, jobGVR); !done {
+		t.Error("resp event didn't hit the target pod")
+		t.Failed()
+	}
+}
+
+func createresource(client *Client, config map[string]string, folders []string, t *testing.T) *Installer {
+	installer := NewInstaller(client.Dynamic, config,
+		EndToEndConfigYaml(folders)...)
+	if err := installer.Do("create"); err != nil {
+		t.Errorf("failed to create, %s", err)
+		return nil
+	}
+	return installer
+}
+
+func deleteResource(installer *Installer, t *testing.T) {
+	if err := installer.Do("delete"); err != nil {
+		t.Errorf("failed to delete, %s", err)
+	}
+	// Just chill for tick.
+	time.Sleep(15 * time.Second)
+}
+
+func jobDone(client *Client, podName string, t *testing.T, jobGVR schema.GroupVersionResource) bool {
```

Comment:
If these are utility methods for all tests, I think we should try to move them to another go file. And use them from there everywhere. The fact that all are in the same package, gives them visibility, but I'd rather not include pubsub stuff in a storage test.
Does it make sense?